packages/components/src/components/RunAction/RunAction.js

```diff
@@ -0,0 +1,139 @@
+/*
+Copyright 2019-2021 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { Component } from 'react';
+import { injectIntl } from 'react-intl';
+import { Button } from 'carbon-components-react';
+import { Play32 as Play, Restart32 as Restart } from '@carbon/icons-react';
+
+class RunAction extends Component {
+  handleAction = event => {
+    event.preventDefault();
+    const {
+      action,
+      getURL,
+      intl,
+      run,
+      runaction,
+      showNotification
+    } = this.props;
+    const { namespace } = run.metadata;
+    let triggerMsg = null;
+    switch (action) {
+      case 'rerun':
+        triggerMsg = intl.formatMessage({
+          id: 'dashboard.rerun.triggered',
+          defaultMessage: 'Triggered rerun'
+        });
+        break;
+      case 'start':
+        triggerMsg = intl.formatMessage({
+          id: 'dashboard.start.triggered',
+          defaultMessage: 'Triggered start'
+        });
+        break;
+      default:
+        break;
+    }
+
+    runaction(run)
+      .then(newRun => {
+        const finalURL = getURL({
+          namespace,
+          name: newRun.metadata.name
+        });
+        showNotification({
+          message: triggerMsg,
+          kind: 'success',
+          logsURL: finalURL
+        });
+      })
+      .catch(error => {
+        let msg = null;
+        switch (action) {
```

Comment:
@AlanGreene so .. I'm actually _not_ a frontend dev, but trying to learn as I go w/ this one.  Is this the best way to manage the intl id's?  It appears they must be static.  If there is a better pattern / example for me to replicate I'm happy to change if you can point me in the right direction.

Reply:
That's right, they must be static. This approach is fine, if we end up with many more actions in future we might consider refactoring it but it's not a problem for now. 👍

Reply:
ok sounds good, I will leave it this way for now